internal/test/test.go

```diff
@@ -191,13 +192,35 @@ func testCode(
 			WithFileResolver(fileResolver(scriptPath, state)).
 			WithContracts(contracts)
 
-		results, err := runner.RunTests(string(code))
-		if err != nil {
-			return nil, err
+		if flags.Name != "" {
+			testFunctions, err := runner.GetTests(string(code))
+			if err != nil {
+				return nil, err
+			}
+			includesTest := false
+			for _, testFunction := range testFunctions {
+				if testFunction == flags.Name {
+					includesTest = true
+					break
+				}
+			}
+
+			if includesTest {
```

Comment:
I believe we can get rid of this `if includesTest` condition, by simply moving its body in the `if testFunction == flags.Name`.

Reply:
You could also reverse the `if testFunction == flags.Name`, e.g.:
```go
if testFunction != flags.Name {
	continue
}

result, err := runner.RunTest(string(code), flags.Name)
if err != nil {
	return nil, err
}
testResults[scriptPath] = []cdcTests.Result{*result}
```

Reply:
This was a workaround for running `flow test` across multiple files.  e.g. `flow test test/**/*.cdc`.

If we don't check for the existence of the test, each file contained in the test run will receive an error from the `runner.RunTest` function, so the idea was to only run the test if it exists

i.e.

{
    "test/foo.cdc": {
        "testFoo": "Error: Test not found......"
    },
    "test/bar.cdc": {
        "testFoo": "Error: Test not found......"
    },
    "test/baz.cdc": {
        "testFoo": "PASS"
    }
}

Reply:
I understand that :sweat_smile: I am not suggesting to alter the logic of it, it was merely a suggestion to eliminate the `bool` variable.

Reply:
Ah sorry, read that too quickly, I misunderstood :).  Yup I can do that for sure 👍